chia/full_node/mempool_check_conditions.py

```diff
@@ -188,19 +192,14 @@ def get_spends_for_block_with_conditions(generator: BlockGenerator, height: int,
             DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM,
             flags,
         )
-        spends.append({
-            "coin": {
-                "amount": int_from_bytes(amount.atom),
-                "parent_coin_info": parent.atom,
-                "puzzle_hash": puzzle_hash,
-            },
-            "puzzle": puzzle,
-            "solution": solution,
-            "conditions": Program.to(r),
-        })
+        coin = Coin(parent.atom, puzzle_hash, int_from_bytes(amount.atom))
+        coin_spend = CoinSpend(coin, puzzle, solution)
+        conditions = Program.to(r)
```

Comment:
If you, instead of `Program` make this variable a `List[ConditionWithArg]`, you can use the existing function: `parse_sexp_to_conditions()`. It does some sanitization